projects/optic/src/__tests__/integration/integration.ts

```diff
@@ -125,3 +127,40 @@ export async function fileExists(path: string): Promise<boolean> {
 export function normalizeWorkspace(workspace: string, text: string): string {
   return text.replace(new RegExp(workspace, 'g'), '$$workspace$$');
 }
+
+let server: http.Server;
+
+type MockHttpHandler = (req: { url: string; method: string }) => any;
```

Comment:
Spun up a web service to respond to mock events - takes in a mock handler which can be used to pass in functions, etc